tests/testCoaddInputs.py

```diff
@@ -0,0 +1,256 @@
+#
+# LSST Data Management System
+# Copyright 2008-2016 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""
+Tests for lsst.afw.table.CoaddInputs
+
+Note: generating the version 1 data requires some complicated machinations
+because CoaddInputRecorderTask had bugs:
+- Setup pipe_tasks 12.1, or a similar version old enough to have version 1 CoaddInput/ExposureTable
+- Edit this file as follows:
+    - Set SaveCoadd = True
+    - Set self.version to 2 in CoaddInputsTestCase
+- Edit CoaddInputRecorderTask as follows:
+    - Apply all fixes to CoaddInputRecorderTask from DM-7976
+    - Comment out the line that saves VisitInfo (since version 1 doesn't have it)
+- Edit ups/pipe_tasks.table as follows:
+    - Do not import obs_base (since it did not exist at that time)
+- "setup -r . 'j" to setup this version of pipe_tasks
+- "python tests/testCoaddInputs.py" to run this test.
+    You will see some errors, but the needed data file will be written:
+    tests/data/testCoaddInputs_coadd_with_version_1_data.fits
+- Save the data file to the repository (but do not save any other changes).
+"""
+
+from __future__ import absolute_import, division, print_function
+import os.path
+import unittest
+
+import numpy as np
+
+import lsst.utils.tests
+import lsst.pex.exceptions
+from lsst.daf.base import DateTime
+import lsst.afw.cameraGeom.testUtils
+from lsst.afw.coord import Coord, IcrsCoord
+import lsst.afw.geom
+from lsst.afw.geom import degrees
+from lsst.afw.geom.polygon import Polygon
+import lsst.afw.image
+from lsst.afw.detection import GaussianPsf
+from lsst.afw.math import ChebyshevBoundedField
+from lsst.pipe.tasks.coaddInputRecorder import CoaddInputRecorderTask
+
+SaveCoadd = False  # if True then save coadd even if test passes (always saved if a test fails)
```

Comment:
Could this lead to problems if the test fails for some new reason, and then confusingly continues to fail because the committed test data file has been overwritten by something incorrect?